dotcom-rendering/src/web/components/ElementContainer.tsx

```diff
@@ -29,12 +29,19 @@ const setBackgroundColour = (colour: string) => emoCss`
 	background-color: ${colour};
 `;
 
+const setBackgroundColourMobile = (colour: string) => emoCss`
+	${until.desktop} {
+		background-color: ${colour};
+	}
+`;
+
 type Props = {
 	sectionId?: string;
 	showSideBorders?: boolean;
 	showTopBorder?: boolean;
 	padded?: boolean;
 	backgroundColour?: string;
+	backgroundColourMobile?: string;
```

Comment:
Thanks for this. I can see your point, but I'm equally concerned about the code duplication this represents. It's an additional 200+ lines of code in `LiveLayout` vs an additional optional prop in `ElementContainer`. It seems like a fairly extreme stance to take in order not to alter the interface of this component. Very happy to discuss though.

Reply:
In my head the duplication would look a bit like this

```ts
<Hide above="leftCol">
	<ElementContainer
		backgroundColour={palette.background.article}
	>
		<SomeVerticalSpace />
	</ElementContainer>
</Hide>
<Hide below="leftCol">
	<ElementContainer
		backgroundColour={palette.background.someOtherProperty}
	>
		<SomeVerticalSpace />
	</ElementContainer>
</Hide>
```

The divider space is just empty so it could just be a plain div.

Is this possible? I'm very aware I don't have a complete picture of the use cases here!

Reply:
Just to understand your point here,
are you suggesting instead of having 2 props `backgroundColour` & `backgroundColourMobile` we can only have `backgroundColour` and the consumer decides the breakpoint colour by hiding and showing element in a specific breakpoint?

Reply:
Yes, exactly as you describe.

The issue with adding `backgroundColourMobile` to the api is it makes it seem like having a different background colour at mobile is a 'thing' that the Guardian does. The intention for this component is to extend its usage, perhaps hoisting it into Source, so the simpler and easier to understand the props are the better